RFS/src/main/java/com/rfs/worker/IndexStep.java

```diff
@@ -0,0 +1,475 @@
+        protected final List<CmsEntry.IndexWorkItem> workItems;
+
+        public MigrateIndices(SharedMembers members, List<CmsEntry.IndexWorkItem> workItems) {
+            super(members);
+            this.workItems = workItems;
+        }
+
+        @Override
+        public void run() {
+            logger.info("Migrating current batch of indices...");
+            for (CmsEntry.IndexWorkItem workItem : workItems) {
+                /*
+                 * Try to migrate the index.
+                 * 
+                 * If we succeed, we forcefully mark it as completed.  When we do so, we don't care if someone else has changed
+                 * the record in the meantime; *we* completed it successfully and that's what matters.  Because this is the only
+                 * forceful operation on the entry, the other operations are safe to be non-forceful.
+                 * 
+                 * If it's already exceeded the number of attempts, we attempt to mark it as failed.  If someone else
+                 * has updated the entry in the meantime, we just move on to the next work item.  This is safe because
+                 * it means someone else has either marked it as completed or failed, and either is fine.
+                 * 
+                 * If we fail to migrate it, we attempt to increment the attempt count.  It's fine if the increment
+                 * fails because we guarantee that we'll attempt the work at least N times, not exactly N times.
+                 */
+                if (workItem.numAttempts > CmsEntry.IndexWorkItem.ATTEMPTS_SOFT_LIMIT) {
+                    logger.info("Index Work Item " + workItem.name + " has exceeded the maximum number of attempts; marking it as failed...");
+                    CmsEntry.IndexWorkItem updatedEntry = new CmsEntry.IndexWorkItem(
+                        workItem.name,
+                        CmsEntry.IndexWorkItemStatus.FAILED,
+                        workItem.numAttempts,
+                        workItem.numShards
+                    );
+
+                    members.cmsClient.updateIndexWorkItem(updatedEntry, workItem).ifPresentOrElse(
+                        value -> logger.info("Index Work Item " + workItem.name + " marked as failed"),
+                        () ->logger.info("Unable to mark Index Work Item " + workItem.name + " as failed")
+                    );
+                    continue;
+                }
+
+                try {
+                    logger.info("Migrating index: " + workItem.name);
+                    IndexMetadata.Data indexMetadata = members.metadataFactory.fromRepo(members.snapshotName, workItem.name);
+
+                    ObjectNode root = indexMetadata.toObjectNode();
+                    ObjectNode transformedRoot = members.transformer.transformIndexMetadata(root);
+
+                    IndexMetadataData_OS_2_11 indexMetadataOS211 = new IndexMetadataData_OS_2_11(transformedRoot, indexMetadata.getId(), workItem.name);
```

Comment:
Lets remove hardcoding the target index type

Reply:
Ah, oops.  I thought I had addressed this.  Yeah, I agree - will do.